### PR TITLE
Add data_prefix check for float limit_val_batches

### PR DESCRIPTION
**Pull request by athitten**
# What does this PR do ?

Set num of eval samples as 1 for float `limit_val_batches` only when the data_prefix contains single prefix, since otherwise it can lead to no validation.

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -3595,14 +3594,15 @@ assert_frame_equal(training_curve, gt_curve, rtol=1e-3, atol=1e-3)"'''
         model.activations_checkpoint_method='block' \
         model.activations_checkpoint_granularity='full' \
         model.activations_checkpoint_num_layers=1 \
+        model.data.data_prefix={"train":[1.0, /home/TestData/nlp/megatron_gpt/data/gpt/simple_wiki_gpt_preproc_text_document], "validation":[/home/TestData/nlp/megatron_gpt/data/gpt/simple_wiki_gpt_preproc_text_document], "test":[/home/TestData/nlp/megatron_gpt/data/gpt/simple_wiki_gpt_preproc_text_document]} \
```
Comment (athitten) [inline Jenkinsfile:3597]: Pass `data_prefix` as a DictConfig to test float `limit_val_batches`.

**Comment by athitten** [comment]
jenkins

**Review by ShriyaPalsamudram — DISMISSED** [review]

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Comment by athitten** [comment]
jenkins

**Pull request closed without merging**